### PR TITLE
fix(node agent): update compatibility requirements with newer features

### PR DESCRIPTION
**Pull request by jordigh**
This closes [#1254 of the node-newrelic repo](https://github.com/newrelic/node-newrelic/issues/1254)

A couple of other updates are also included for other instrumented libraries.

### CODE REVIEW

**Review by homelessbirds — APPROVED** [review]

File: src/content/docs/apm/agents/nodejs-agent/getting-started/compatibility-requirements-nodejs-agent.mdx [hunk 1]
```diff
@@ -325,6 +325,21 @@ For other message queue libraries, use [custom instrumentation](/docs/agents/nod
 * [Microsoft Azure](/docs/nodejs/nodejs-agent-on-microsoft-azure)
 * [Heroku](/docs/nodejs/nodejs-agent-on-heroku)
 
+## HTTP/1.1
+
+The Node.js agent instruments the native `http` module. In addition, the popular [undici](https://www.npmjs.com/package/undici) client is also supported.
```
Comment (bizob2828) [inline src/content/docs/apm/agents/nodejs-agent/getting-started/compatibility-requirements-nodejs-agent.mdx:330]: as mentioned in standup.  currently undici is not generally available.  you must set `feature_flag.undici_instrumentation = true`

File: src/content/docs/apm/agents/nodejs-agent/getting-started/compatibility-requirements-nodejs-agent.mdx [hunk 2]
```diff
@@ -158,12 +158,12 @@ Errors resulting in unhandled rejections are not scoped to the transaction that
 * [Hapi](https://www.npmjs.com/package/hapi)
 * [Koa](https://www.npmjs.com/package/koa) 2.0.0 or higher ([external module](https://github.com/newrelic/node-newrelic-koa) loaded with the agent)
 * [Fastify](https://www.npmjs.com/package/fastify)
-* [Next.js](https://www.npmjs.com/package/next) 12.0.9 or higher
+* [Next.js](https://www.npmjs.com/package/next) 12.0.9 or higher (12.2.0 or higher required for middleware instrumentation)
```
Comment (bizob2828) [inline src/content/docs/apm/agents/nodejs-agent/getting-started/compatibility-requirements-nodejs-agent.mdx:161]: ooh, good point

File: src/content/docs/apm/agents/nodejs-agent/getting-started/compatibility-requirements-nodejs-agent.mdx [hunk 3]
```diff
@@ -325,6 +325,21 @@ For other message queue libraries, use [custom instrumentation](/docs/agents/nod
 * [Microsoft Azure](/docs/nodejs/nodejs-agent-on-microsoft-azure)
 * [Heroku](/docs/nodejs/nodejs-agent-on-heroku)
 
+## HTTP/1.1
+
+The Node.js agent instruments the native `http` module. In addition, the popular [undici](https://www.npmjs.com/package/undici) client is also supported.
+
+## gRPC
+
+* The [grpc-js](https://www.npmjs.com/package/@grpc/grpc-js) library is supported for unary, streaming, or bidirectional client calls.
```
Comment (bizob2828) [inline src/content/docs/apm/agents/nodejs-agent/getting-started/compatibility-requirements-nodejs-agent.mdx:334]: 🎉

**Review by bizob2828 — CHANGES_REQUESTED** [review]

File: src/content/docs/apm/agents/nodejs-agent/getting-started/compatibility-requirements-nodejs-agent.mdx [hunk 4]
```diff
@@ -325,6 +325,21 @@ For other message queue libraries, use [custom instrumentation](/docs/agents/nod
 * [Microsoft Azure](/docs/nodejs/nodejs-agent-on-microsoft-azure)
 * [Heroku](/docs/nodejs/nodejs-agent-on-heroku)
 
+## HTTP/1.1
+
+The Node.js agent instruments the native `http` module. In addition, the popular [undici](https://www.npmjs.com/package/undici) client also has experimental support.
```
Comment (akristen) [inline src/content/docs/apm/agents/nodejs-agent/getting-started/compatibility-requirements-nodejs-agent.mdx:330]:
```suggestion
The Node.js agent instruments the native `http` module. 

In addition, the popular [undici](https://www.npmjs.com/package/undici) client has experimental support as well. To use undici, be sure to set `feature_flag.undici_instrumentation = true` in the `newrelic.js` [configuration file](/docs/apm/agents/nodejs-agent/installation-configuration/nodejs-agent-configuration/#methods-and-precedence).
```

**Review by akristen — APPROVED** [review]

**Review by akristen — APPROVED** [review]
Thanks for your help + patience on this! Excited to merge it in

**Pull request merged**